src/main/java/duke/functions/Duke.java

```diff
@@ -0,0 +1,288 @@
+                } catch (WrongFormatException e) {
+                    System.out.println("\tWrong format! Try \"Deadline [description] \\by [due date]\"");
+                }
+                try {
+                    writeToFile(items);
+                } catch (IOException e) {
+                    System.out.println("\tError writing to file");
+                }
+                break;
+            }
+            case "event": {
+                try {
+                    handleEvent(items, userInput);
+                } catch (EmptyArgException e) {
+                    System.out.println("\tDescription for deadline cannot be empty!");
+                } catch (WrongFormatException e) {
+                    System.out.println("\tWrong format! Try \"Event [description] \\by [due date]\"");
+                }
+                try {
+                    writeToFile(items);
+                } catch (IOException e) {
+                    System.out.println("\tError writing to file");
+                }
+                break;
+            }
+            default:
+                System.out.println("\tI don't know what that means");
+                break;
+            }
+        }
+
+        printBye();
+    }
+
+    private static void writeToFile(ArrayList<Task> items) throws IOException {
+        FileWriter fw = new FileWriter(filepath);
+        for (Task i : items) {
+            if (i instanceof Todo) {
+                fw.write("T" + "--" + i.getStatusIcon() + "--" + i.description);
+                fw.write(System.lineSeparator());
+            } else if (i instanceof Deadline) {
+                fw.write("D" + "--" + i.getStatusIcon() + "--" + i.description + "--" + ((Deadline) i).getBy());
+                fw.write(System.lineSeparator());
+
+            } else if (i instanceof Event) {
+                fw.write("E" + "--" + i.getStatusIcon() + "--" + i.description + "--" + ((Event) i).getAt());
+                fw.write(System.lineSeparator());
+            }
+        }
+        if (items.isEmpty()){
+            fw.write("");
+        }
+        fw.close();
+    }
+
+    private static void readSavedList(ArrayList<Task> items) throws IOException {
+        File f = new File(filepath);
+        if (f.createNewFile()) {
+            drawLine();
+            System.out.println("\tNo saved task list, new file created");
+            drawLine();
+        } else {
+            Scanner fs = new Scanner(f);
+            while (fs.hasNext()) {
+                String[] feed = fs.nextLine().split("--");
+
+                switch (feed[0]) {
+                case "T":
+                    items.add(new Todo(feed[2]));
+                    break;
+                case "D":
+                    items.add(new Deadline(feed[2], feed[3]));
+                    break;
+                case "E":
+                    items.add(new Event(feed[2], feed[3]));
+                    break;
+                default:
+                    break;
```

Comment:
what does feed[0],[2],[3] stands for? perhaps improve the readability (e.g feed[0]-> taskType)?